plugins/woocommerce/src/StoreApi/Legacy.php

```diff
@@ -56,16 +56,24 @@ public function process_legacy_payment( PaymentContext $context, PaymentResult &
 		// Restore $_POST data.
 		$_POST = $post_data;
 
-		// If `process_payment` added notices, clear them. Notices are not displayed from the API -- payment should fail,
-		// and a generic notice will be shown instead if payment failed.
-		wc_clear_notices();
+		// If the payment failed with a message, throw an exception.
+		if ( 'failure' === $gateway_result['result'] ) {
```

Comment:
This will create a PHP error notice if `result` does not exist in the array. Since we handle this further down (`$result_status = $gateway_result['result'] ?? 'failure';`) you could move that line above this and use that.

Reply:
I will check for that instead of moving that line. That check will always assign `failure` as a result, and we can't know if the payment method explicitly returned failure or just didn't do anything.